System.Containers/ContainerHelpers.cs

```diff
@@ -0,0 +1,97 @@
+namespace System.Containers
+{
+    public static class ContainerHelpers
+    {
+        private static string[] knownImageSeparators = { "-", ".", "_" };
+
+        /// <summary>
+        /// Given some "fully qualified" image name (e.g. mcr.microsoft.com/dotnet/runtime), return
+        /// a valid UriBuilder. This means appending 'https' if the URI is not absolute, otherwise UriBuilder will throw.
+        /// </summary>
+        /// <param name="containerBase"></param>
+        /// <returns>A UriBuilder with the given containerBase, or, if containerBase is relative, https:// + containerBase</returns>
+        private static UriBuilder? ContainerImageToUriBuilder(string containerBase)
+        {
+            Uri uri = new Uri(containerBase, UriKind.RelativeOrAbsolute);
+
+            try
+            {
+                if (uri.IsAbsoluteUri)
+                {
+                    return new UriBuilder(uri);
+                }
+                else
+                {
+                    // todo: allow customization?
+                    return new UriBuilder(containerBase.Contains("localhost") ? "http://" : "https://" + uri);
+                }
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("Failed parsing the container image into a UriBuilder: {0}", e);
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Ensures the given image does not start or end with known separators, which are invalid.
+        /// </summary>
+        /// <param name="imageName"></param>
+        /// <returns></returns>
+        public static bool IsValidImageName(string imageName)
+        {
+            for (int i = 0; i < knownImageSeparators.Length; i++)
+            {
+                if (imageName.StartsWith(knownImageSeparators[i]) || imageName.EndsWith(knownImageSeparators[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Ensures the given tag is less than 129 characters, and does not start with invalid characters.
```

Comment:
Link to the spec please https://github.com/opencontainers/distribution-spec/blob/4ab4752c3b86a926d7e5da84de64cbbdcc18d313/spec.md#pulling-manifests